src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -138,10 +138,10 @@ public string LinkName
         /// <remarks>The value in this field has no effect on Windows platforms.</remarks>
         public UnixFileMode Mode
         {
-            get => (UnixFileMode)_header._mode;
+            get => (UnixFileMode)(_header._mode & 0xFFF); // mask to only keep least significant 12-bits
             set
             {
-                if ((int)value is < 0 or > 4095) // 4095 in decimal is 7777 in octal
+                if ((int)value is < 0 or > 0xFFF) // 0xFFF (4095) in decimal is 7777 in octal
```

Comment:
Same here. Can the comment be updated to reflect we're checking for valid values of `UnixFileMode`?
I've written this check elsewhere as `(value & ~FileSystem.ValidUnixFileModes) != 0)`.